src/services/vault-connector.ts

```diff
@@ -38,30 +38,36 @@ export class VaultConnector {
       sealedcode: 200,
       uninitcode: 200,
     });
-    if (!health.initialized) {
-      const resp = await this._vaultClient.init(opts);
-      // eslint-disable-next-line @typescript-eslint/prefer-optional-chain
-      if (!resp || !resp.keys || resp.keys.length === 0 || !resp.root_token) {
-        console.error(`Vault initialisation failed ! Response :: ${resp}`);
-        throw new Error('Vault initialisation failed !');
+    switch (true) {
```

Comment:
using switch is a bit vague here, because general use cases have a variable passed in here and values inside the cases.

Reply:
resolved